### PR TITLE
use jsoniter for json marshal/unmarshal in KMS

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
use jsoniter for json marshal/unmarshal in KMS

## Motivation and Context
as an optimization, we are okay to tax
the legacy objects - fallback only when
json.Unmarshal fails.

## How to test this PR?
Nothing special its an optimization 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :x: [more...](#30ef550c-68d0-4c44-a68e-2192cee8f78e) |
| mint-gateway-nas.sh | :x: [more...](#38974e7f-ef05-43b2-9aad-035701066fcb) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#1e6ae5c5-dbff-4ad4-a482-5b36f5b76270) |
<a name="30ef550c-68d0-4c44-a68e-2192cee8f78e"></a>
### 12146-efb19f8/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31975
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d38cc766e066:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 2 minutes and 20 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 20 seconds
(8/15) Running minio-dotnet tests ... done in 49 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 56 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 52 seconds
(11/15) Running minio-js tests ... done in 48 seconds
(12/15) Running minio-py tests ... FAILED in 2 minutes and 37 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a",
    "object_name": "ad01dd84-25dd-406c-ad71-10fa538c7a7a"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a, request_id: 1678F9B3BA9AE08E, host_id: 799b2f07-3cbd-4b27-9045-042c890f5d08, bucket_name: minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a, request_id: 1678F9B3BA9AE08E, host_id: 799b2f07-3cbd-4b27-9045-042c890f5d08, bucket_name: minio-py-test-853be951-5d39-48c7-9751-dfefc2a4df0a\n",
  "duration": 5488
}
(12/15) Running s3cmd tests ... done in 18 seconds
(13/15) Running s3select tests ... done in 6 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="38974e7f-ef05-43b2-9aad-035701066fcb"></a>
### 12146-efb19f8/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31819
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 788d2b65a48b:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 2 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 45 seconds
(8/15) Running minio-dotnet tests ... done in 35 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-tidwoxlrju161ci0",
    "objectName": "g1jj3uffaxd1grwj1jccr5xubbeh16"
  },
  "duration": 6820,
  "error": "read tcp 172.17.0.8:34614->72.28.97.59:31819: read: connection reset by peer",
  "function": "GetObject(bucketName, objectName)",
  "message": "CopyN failed",
  "name": "minio-go: testSSES3EncryptedGetObjectReadSeekFunctional",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 12 seconds
(10/15) Running minio-js tests ... done in 40 seconds
(11/15) Running minio-py tests ... done in 1 minutes and 14 seconds
(12/15) Running s3cmd tests ... done in 14 seconds
(13/15) Running s3select tests ... done in 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="1e6ae5c5-dbff-4ad4-a482-5b36f5b76270"></a>
### 12146-efb19f8/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31846
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp a8e7587bba27:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 31 seconds
{
  "name": "awscli",
  "duration": 2671,
  "function": "aws --endpoint-url http://minio-c3.minio.io:31846 s3api copy-object --bucket awscli-mint-test-bucket-2140 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-2140/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 44 seconds
(7/15) Running minio-dotnet tests ... done in 35 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 13 seconds
{
  "args": {},
  "duration": 409,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 2 minutes and 41 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 139,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-3n0quqb, objectName = minio-java-test-19qcdlb-copy, resource = /minio-java-test-3n0quqb/minio-java-test-19qcdlb-copy, requestId = 1678F96E77B2830F, hostId = 022edaa5-9213-4173-a347-dfe576152828)\nrequest={method=PUT, url=http://minio-c3.minio.io:31846/minio-java-test-3n0quqb/minio-java-test-19qcdlb-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-36u3qqq/minio-java-test-19qcdlb\nHost: minio-c3.minio.io:31846\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210425T023838Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210425/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 418\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Sun, 25 Apr 2021 02:38:38 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 1678F96E77B2830F\nX-Xss-Protection: 1; mode=block\nDate: Sun, 25 Apr 2021 02:38:38 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 46 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 40 seconds
(10/15) Running s3cmd tests ... FAILED in 5 seconds
{
  "name": "s3cmd",
  "duration": "2821",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-30301/s3cmd-test-object-31736 8d22bab5-bb0b-4e69-83ed-14476446c0da\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 8d22bab5-bb0b-4e69-83ed-14476446c0da put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
```
Error running mint automation
+ mkdir -p 12146-b11076a/gopath/src/github.com/minio
+ git -C 12146-b11076a/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 12146-b11076a/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 12146-b11076a/gopath/src/github.com/minio/minio fetch --quiet minio pull/12146/head:pr12146
+ git -C 12146-b11076a/gopath/src/github.com/minio/minio checkout --quiet pr12146
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/12146-b11076a/gopath
+ make -C 12146-b11076a/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/Azure/azure-pipeline-go v0.2.2
go: downloading github.com/dustin/go-humanize v1.0.0
go: downloading github.com/minio/cli v1.22.0
go: downloading google.golang.org/api v0.25.0
go: downloading github.com/minio/minio-go/v7 v7.0.11-0.20210302210017-6ae69c73ce78
go: downloading github.com/colinmarc/hdfs/v2 v2.2.0
go: downloading github.com/jcmturner/gokrb5/v8 v8.4.2
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading cloud.google.com/go v0.57.0
go: downloading github.com/Azure/azure-storage-blob-go v0.10.0
go: downloading github.com/minio/sha256-simd v1.0.0
go: downloading github.com/minio/highwayhash v1.0.2
go: downloading github.com/montanaflynn/stats v0.5.0
go: downloading golang.org/x/sys v0.0.0-20210225134936-a50acf3fe073
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading github.com/secure-io/sio-go v0.3.1
go: downloading github.com/fatih/color v1.10.0
go: downloading golang.org/x/crypto v0.0.0-20210415154028-4f45737414dc
go: downloading github.com/rjeczalik/notify v0.9.2
go: downloading github.com/minio/sio v0.2.1
go: downloading github.com/shirou/gopsutil/v3 v3.21.1
go: downloading github.com/tinylib/msgp v1.1.3
go: downloading golang.org/x/net v0.0.0-20210226172049-e18ecbb05110
go: downloading gopkg.in/yaml.v2 v2.3.0
go: downloading github.com/dswarbrick/smart v0.0.0-20190505152634-909a45200d6d
go: downloading github.com/ncw/directio v1.0.5
go: downloading github.com/VividCortex/ewma v1.1.1
go: downloading github.com/google/uuid v1.1.2
go: downloading github.com/cheggaaa/pb v1.0.29
go: downloading github.com/bcicen/jstream v1.0.1
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/djherbis/atime v1.0.0
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading github.com/gorilla/handlers v1.5.1
go: downloading github.com/klauspost/pgzip v1.2.5
go: downloading github.com/dchest/siphash v1.2.1
go: downloading github.com/klauspost/compress v1.11.12
go: downloading github.com/klauspost/reedsolomon v1.9.11
go: downloading github.com/gorilla/mux v1.8.0
go: downloading github.com/miekg/dns v1.1.35
go: downloading github.com/minio/selfupdate v0.3.1
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/philhofer/fwd v1.1.1
go: downloading github.com/pierrec/lz4 v2.5.2+incompatible
go: downloading github.com/prometheus/client_golang v1.8.0
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading github.com/prometheus/procfs v0.6.0
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading go.etcd.io/etcd v0.0.0-20201125193152-8a03d2e9614b
go: downloading github.com/coredns/coredns v1.4.0
go: downloading github.com/beevik/ntp v0.3.0
go: downloading github.com/go-ldap/ldap/v3 v3.2.4
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/minio/simdjson-go v0.2.1
go: downloading go.uber.org/zap v1.14.1
go: downloading github.com/Shopify/sarama v1.27.2
go: downloading github.com/eclipse/paho.mqtt.golang v1.3.0
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/lib/pq v1.9.0
go: downloading github.com/nats-io/nats.go v1.10.0
go: downloading github.com/nats-io/stan.go v0.8.3
go: downloading github.com/nsqio/go-nsq v1.0.8
go: downloading github.com/olivere/elastic/v7 v7.0.22
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/streadway/amqp v1.0.0
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/alecthomas/participle v0.2.1
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/golang/protobuf v1.4.3
go: downloading github.com/felixge/httpsnoop v1.0.1
go: downloading github.com/Azure/go-ntlmssp v0.0.0-20200615164410-66371956d46c
go: downloading github.com/go-asn1-ber/asn1-ber v1.5.1
go: downloading github.com/klauspost/cpuid v1.3.1
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/willf/bitset v1.1.11
go: downloading github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/tidwall/gjson v1.6.8
go: downloading github.com/tidwall/sjson v1.0.4
go: downloading github.com/nats-io/nuid v1.0.1
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/golang/snappy v0.0.1
go: downloading github.com/xdg/stringprep v1.0.0
go: downloading github.com/gorilla/websocket v1.4.2
go: downloading github.com/prometheus/common v0.14.0
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/tidwall/pretty v1.0.2
go: downloading github.com/tidwall/match v1.0.3
go: downloading cloud.google.com/go/storage v1.8.0
go: downloading github.com/nats-io/nkeys v0.2.0
go: downloading github.com/nats-io/jwt v1.1.0
go: downloading golang.org/x/text v0.3.3
go: downloading go.uber.org/atomic v1.6.0
go: downloading go.uber.org/multierr v1.5.0
go: downloading github.com/hashicorp/go-uuid v1.0.2
go: downloading github.com/jcmturner/gofork v1.0.0
go: downloading github.com/jcmturner/dnsutils/v2 v2.0.0
go: downloading github.com/jcmturner/aescts/v2 v2.0.0
go: downloading github.com/jcmturner/rpc/v2 v2.0.3
go: downloading google.golang.org/protobuf v1.24.0
go: downloading gopkg.in/ini.v1 v1.57.0
go: downloading github.com/minio/md5-simd v1.1.1
go: downloading github.com/rs/xid v1.2.1
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/klauspost/cpuid/v2 v2.0.4
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.5.0
go: downloading github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/rcrowley/go-metrics v0.0.0-20200313005456-10cdbea86bc0
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading google.golang.org/genproto v0.0.0-20200527145253-8367513e4ece
go: downloading github.com/googleapis/gax-go/v2 v2.0.5
go: downloading google.golang.org/grpc v1.29.1
go: downloading go.opencensus.io v0.22.5
go: downloading github.com/jcmturner/goidentity/v6 v6.0.1
go: downloading golang.org/x/oauth2 v0.0.0-20200107190931-bf48bf16ab8d
go: downloading github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
go: downloading github.com/mailru/easyjson v0.7.6
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/coreos/go-semver v0.3.0
go: downloading github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f
go: downloading github.com/coreos/go-systemd v0.0.0-20190321100706-95778dfbb74e
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
# github.com/minio/minio/pkg/kms
pkg/kms/single-key.go:187:13: kms.legacyDecryptKey undefined (type secretKey has no field or method legacyDecryptKey)
make: *** [build] Error 2

```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-zoned.sh | :x: [more...](#6c7722fb-45c4-4321-aabb-fe3634924331) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#9be315f6-33f6-4cdc-a0b0-5734603edc2d) |
<a name="6c7722fb-45c4-4321-aabb-fe3634924331"></a>
### 12146-6f125f7/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev8.minio.io:30714
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5d28d7f2f949:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... done in 2 minutes and 16 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 11 seconds
(8/15) Running minio-dotnet tests ... done in 49 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 58 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 47 seconds
(11/15) Running minio-js tests ... done in 48 seconds
(12/15) Running minio-py tests ... FAILED in 2 minutes and 40 seconds
{
  "name": "minio-py:test_thread_safe",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-d918f10f-5707-424c-8d5e-beabfa98e32f",
    "object_name": "9b550b77-bed6-410f-835b-97e6cfedd0b8"
  },
  "message": "S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-d918f10f-5707-424c-8d5e-beabfa98e32f, request_id: 16792D27F7CC47D3, host_id: 20b4c764-8c91-4e16-bc17-d33b39680a1d, bucket_name: minio-py-test-d918f10f-5707-424c-8d5e-beabfa98e32f",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1640, in test_thread_safe\n    _CLIENT.remove_bucket(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 676, in remove_bucket\n    self._execute(\"DELETE\", bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 409, in _execute\n    preload_content=preload_content,\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 385, in _url_open\n    raise response_error\nminio.error.S3Error: S3 operation failed; code: BucketNotEmpty, message: The bucket you tried to delete is not empty, resource: /minio-py-test-d918f10f-5707-424c-8d5e-beabfa98e32f, request_id: 16792D27F7CC47D3, host_id: 20b4c764-8c91-4e16-bc17-d33b39680a1d, bucket_name: minio-py-test-d918f10f-5707-424c-8d5e-beabfa98e32f\n",
  "duration": 6893
}
(12/15) Running s3cmd tests ... done in 19 seconds
(13/15) Running s3select tests ... done in 5 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="9be315f6-33f6-4cdc-a0b0-5734603edc2d"></a>
### 12146-6f125f7/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31344
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 777b6c8f714b:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 43 seconds
(4/15) Running aws-sdk-ruby tests ... done in 3 seconds
(5/15) Running awscli tests ... FAILED in 32 seconds
{
  "name": "awscli",
  "duration": 2710,
  "function": "aws --endpoint-url http://minio-c3.minio.io:31344 s3api copy-object --bucket awscli-mint-test-bucket-15880 --key datafile-1-kB-copy --copy-source awscli-mint-test-bucket-15880/datafile-1-kB\n",
  "status": "FAIL",
  "error": "Hash mismatch expected 084e1383b70fb0c51acc680fef370023, got ac57de7156d7fc25ac1a65f81fa3989b"
}
(5/15) Running healthcheck tests ... done in 1 seconds
(6/15) Running mc tests ... done in 48 seconds
(7/15) Running minio-dotnet tests ... done in 37 seconds
(8/15) Running minio-go tests ... FAILED in 2 minutes and 14 seconds
{
  "args": {},
  "duration": 364,
  "error": "At least one of the pre-conditions you specified did not hold",
  "function": "CopyObjectPart(destination, source)",
  "message": "CopyObjectPart call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... FAILED in 2 minutes and 15 seconds
{
  "name": "minio-java",
  "function": "copyObject()",
  "args": "[match etag]",
  "duration": 120,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = PreconditionFailed, message = At least one of the pre-conditions you specified did not hold, bucketName = minio-java-test-8776ii, objectName = minio-java-test-1110sfh-copy, resource = /minio-java-test-8776ii/minio-java-test-1110sfh-copy, requestId = 16792CE19341F429, hostId = 9ac52fd4-f4a0-47b9-8e66-3a1449fb0a37)\nrequest={method=PUT, url=http://minio-c3.minio.io:31344/minio-java-test-8776ii/minio-java-test-1110sfh-copy, headers=x-amz-copy-source-if-match: 71cff0a060f852067e443ad1e24ae26c-1\nx-amz-copy-source: /minio-java-test-29q16ln/minio-java-test-1110sfh\nHost: minio-c3.minio.io:31344\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20210425T182128Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210425/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-copy-source;x-amz-copy-source-if-match;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=412, headers=Accept-Ranges: bytes\nContent-Length: 416\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nETag: \"71cff0a060f852067e443ad1e24ae26c\"\nLast-Modified: Sun, 25 Apr 2021 18:21:28 GMT\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 16792CE19341F429\nX-Xss-Protection: 1; mode=block\nDate: Sun, 25 Apr 2021 18:21:28 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.execute(MinioClient.java:563), io.minio.MinioClient.executePut(MinioClient.java:904), io.minio.MinioClient.copyObject(MinioClient.java:1232), FunctionalTest.testCopyObjectMatchETag(FunctionalTest.java:1850), FunctionalTest.copyObject(FunctionalTest.java:2016), FunctionalTest.runObjectTests(FunctionalTest.java:3757), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(8/15) Running minio-js tests ... done in 48 seconds
(9/15) Running minio-py tests ... done in 2 minutes and 48 seconds
(10/15) Running s3cmd tests ... FAILED in 6 seconds
{
  "name": "s3cmd",
  "duration": "3596",
  "function": "test_put_object_multipart",
  "status": "FAIL",
  "error": "WARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Retrying upload of /mint/data/datafile-65-MB\nWARNING: MD5 Sums don't match!\nWARNING: Too many failures. Giving up on '/mint/data/datafile-65-MB'\nERROR: \nUpload of '/mint/data/datafile-65-MB' part 1 failed. Use\n  /usr/local/bin/s3cmd abortmp s3://s3cmd-test-bucket-12480/s3cmd-test-object-14819 51039d56-97e4-4a1e-bdef-8ff197d16a8b\nto abort the upload, or\n  /usr/local/bin/s3cmd --upload-id 51039d56-97e4-4a1e-bdef-8ff197d16a8b put ...\nto continue the upload.\nERROR: Upload of '/mint/data/datafile-65-MB' failed too many times (Last reason: )"
}
(10/15) Running s3select tests ... done in 6 seconds
(11/15) Running security tests ... done in 0 seconds

Executed 11 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/erasure-object.go [hunk 1]
```diff
@@ -1499,7 +1499,6 @@ func (er erasureObjects) restoreTransitionedObject(ctx context.Context, bucket s
 	}
 
 	_, err = er.CompleteMultipartUpload(ctx, bucket, object, uploadID, uploadedParts, ObjectOptions{
-		VersionID:        oi.VersionID,
```
Comment (klauspost) [inline cmd/erasure-object.go:1502]: Unintentional change? Doesn't seem related.

Comment (harshavardhana) [reply cmd/erasure-object.go:1502]: It is intentional - it is superfluous and not needed.

Comment (harshavardhana) [reply cmd/erasure-object.go:1502]: But I can remove it from this PR.

**Review by aead — APPROVED** [review]
